lxd/storage_dir.go

```diff
@@ -54,19 +54,28 @@ func (s *storageDir) StoragePoolCreate() error {
 	if source == "" {
 		source = filepath.Join(shared.VarPath("storage-pools"), s.pool.Name)
 		s.pool.Config["source"] = source
+	} else {
+		cleanSource := filepath.Clean(source)
+		lxdDir := shared.VarPath()
+		poolMntPoint := getStoragePoolMountPoint(s.pool.Name)
+		if strings.HasPrefix(cleanSource, lxdDir) && cleanSource != poolMntPoint {
+			return fmt.Errorf("DIR storage pool requests in LXD directory \"%s\" are only valid under \"%s\"\n(e.g. source=%s)", shared.VarPath(), shared.VarPath("storage-pools"), poolMntPoint)
+		}
 	}
 
-	err := os.MkdirAll(source, 0711)
-	if err != nil {
-		return err
-	}
 	revert := true
-	defer func() {
-		if !revert {
-			return
+	if !shared.PathExists(source) {
+		err := os.MkdirAll(source, 0711)
+		if err != nil {
+			return err
 		}
-		os.RemoveAll(source)
-	}()
+		defer func() {
+			if !revert {
+				return
+			}
+			os.RemoveAll(source)
```

Comment:
I'd feel better if this was just "os.Remove", I don't like randomly recursively deleting stuff on error :)